packages/astro/src/astro-components/unstyled/SignOutButton.astro

```diff
@@ -0,0 +1,27 @@
+---
+import type { HTMLTag, Polymorphic } from 'astro/types'
+type Props<Tag extends HTMLTag> = Polymorphic<{ as: Tag; id?: string; redirectUrl?: string }>
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
+
+const {
+  as: Tag = 'button',
+  id = safeId,
+  redirectUrl,
+  ...elementProps
+} = Astro.props
+---
+
+<Tag {...elementProps} id={id}>
+  <slot>Sign out</slot>
+</Tag >
+
+<script is:inline define:vars={{ redirectUrl, safeId }}>
+  const btn = document.querySelector(`#${safeId}`);
+
+  btn.addEventListener("click", () => {
+    window.Clerk.signOut({ redirectUrl })
+  });
```

Comment:
Nice catch. I think we need to respect the props like `sessionId` since it's required for signing out of a specific session.

Also, looking at the `SignOutButton` [reference](https://clerk.com/docs/components/unstyled/sign-out-button#sign-out-button-properties), the callback is never stated and it's recommended to migrate to `redirectUrl` in the [Core 2 migration doc](https://clerk.com/docs/upgrade-guides/core-2/remix#other-breaking-changes).

Reply:
I'm still waiting for a response to this but the `signOutCallback` is being deprecated